### PR TITLE
build: Add plugin to check conventional commit message format

### PR DESCRIPTION
**Pull request by ManonPolle**
## Changes description
Add commit checker plugin https://github.com/nicolasfara/conventional-commits
It is based on this https://www.conventionalcommits.org/en/v1.0.0/#summary

**Examples of success commit** 
"feat(buttons): Add a new size for buttons" 
"fix: Fix all ui tests"

**Example of fail commit**
"new: Add my new feature" ( not a conventionnal type ) 
"feat(button): Add a new size for buttons" (  button instead of button**s** is not one of our modules)
": New feature" (no type) 

## Context
Keep a conventional & consistent format in our history. 

## Checklist

- x ] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.

### CODE REVIEW

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/336387c5-46cf-479a-a998-2d401827e9f8/artifact/b15a419871130334/p/05584c5bbf982aea2544f255d02243ed

File: build.gradle.kts [hunk 1]
```diff
@@ -86,6 +87,15 @@ sonarqube {
     }
 }
 
+conventionalCommits {
+    scopes = subprojects.map { it.name }.toList()
```
Comment (Smoochibi) [inline build.gradle.kts:91]: You might want to explicit the `it` variable.

File: build.gradle.kts [hunk 2]
```diff
@@ -86,6 +87,15 @@ sonarqube {
...
+    failureMessage = "Error : The commit message does not meet the Conventional Commit standard\n" +
+            "An example of a valid message is:\n" +
+            "feat(buttons): add the small size button\n" +
+            "Available scopes : ${scopes}\n" +
+            "More details at: https://www.conventionalcommits.org/en/v1.0.0/#summary"
+}
```
Comment (Smoochibi) [inline build.gradle.kts:97]: I see that you are using the `scopes`. 
Is there any particular reason for not using the `types`?

Comment (ManonPolle) [reply build.gradle.kts]: We do are using the types, I just didn't listed in the error message as they are the conventional ones listed in https://www.conventionalcommits.org/en/v1.0.0/#summary

File: build.gradle.kts (hunk 1 above)
Comment (ManonPolle) [reply build.gradle.kts]: You're right, I explicited it

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/485e085c-9f87-4d37-8e7d-f16b02a1a94c/artifact/ef82b4dbd7a2b89a/p/b0c7f0a5582353798899716ef1b29f1d

**Review by Smoochibi — APPROVED** [review]

**Review by GerardPaligot — APPROVED** [review]

**Pull request merged**